File: backend/app/schemas/animal.py
# backend/app/schemas/animal.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import date
from enum import Enum


class SexeEnum(str, Enum):
    MALE = "male"
    FEMELLE = "femelle"
    HERMAPHRODITE = "hermaphrodite"


class StatutAnimalEnum(str, Enum):
    VIVANT = "vivant"
    VENDU = "vendu"
    DECEDE = "decede"
    TRANSFERE = "transfere"


class AnimalBase(BaseModel):
    type_espece: str
    race: str = Field(..., min_length=1, max_length=100)
    sexe: SexeEnum
    date_naissance: Optional[date] = None
    date_arrivee: date
    provenance: Optional[str]
    prix_achat: Optional[float] = None
    enclos_id: int
    statut: StatutAnimalEnum = StatutAnimalEnum.VIVANT
    type_production: Optional[str] = None
    photo_url: Optional[str] = None
    notes: Optional[str] = None

    # === NOUVEAUX CHAMPS ===
    prix_vente: Optional[int] = Field(None, ge=0, description="Prix de vente en €")
    date_vente: Optional[date] = Field(None, description="Date de vente")
    client_acheteur: Optional[str] = Field(None, max_length=255, description="Nom du client acheteur")
    note_vente: Optional[str] = Field(None, description="Notes sur la vente")

class AnimalVenteStats(BaseModel):
    # === NOUVEAU ===
    total_ventes: int
    montant_total_ventes: int
    prix_vente_moyen: int

# === NOUVEAU SCHÉMA POUR LA VENTE ===
class AnimalVenteCreate(BaseModel):
    """Schéma pour enregistrer une vente de bovin, ovin, caprin"""
    prix_vente: int = Field(..., ge=0, description="Prix de vente en FCFA")
    date_vente: Optional[date] = Field(None, description="Date de vente (par défaut aujourd'hui)")
    client_acheteur: Optional[str] = Field(None, max_length=255, description="Nom du client acheteur")
    note_vente: Optional[str] = Field(None, description="Notes sur la vente")
    statut: Optional[StatutAnimalEnum] = Field(StatutAnimalEnum.VENDU, description="Statut après vente")

File: frontend/src/lib/types/animal.ts
// lib/types/animal.ts
export type Sexe = 'male' | 'femelle' | 'hermaphrodite';
export type StatutAnimal = 'vivant' | 'vendu' | 'decede' | 'transfere';

export interface AnimalBase {
    type_espece: string;
    race: string;
    sexe: Sexe;
    date_naissance?: string;
    date_arrivee: string;
    provenance: string;
    prix_achat?: number;
    enclos_id: number;
    enclos_name: string;
    statut: StatutAnimal;
    type_production?: string;
    photo_url?: string;
    notes?: string;

    // === NOUVEAUX CHAMPS DE VENTE ===
    prix_vente?: number | null;
    date_vente?: string | null;
    client_acheteur?: string | null;
    note_vente?: string | null;
}

// Interface pour les filtres
export interface AnimauxFilters {
    searchQuery: string;
    selectedRace: string;
    selectedEnclos: string | number;
    selectedStatuts: StatutAnimal[];
    selectedSexes: Sexe[];
}
export interface LengthResponse{
    users_length: number;
    enclos_length: number;
    bovins_length: number;
    ovins_length: number;
    caprins_length: number;
    avicoles_length: number;
    piscicoles_length: number;
    ruches_length: number;
    nids_length: number;
}

// === NOUVEAU TYPE POUR L'ENREGISTREMENT DE VENTE ===
export interface AnimalVenteStats {
    total_ventes: number;
    montant_total_ventes: number;
    prix_vente_moyen: number;
}

// === NOUVEAU TYPE POUR L'ENREGISTREMENT DE VENTE ===
export interface AnimalVenteCreate {
    prix_vente: number;
    date_vente?: string | null;
    client_acheteur?: string | null;
    note_vente?: string | null;
    statut?: string;
}